lib/pbio/drv/usb/stm32_usbd/usbd_pybricks.c

```diff
@@ -0,0 +1,512 @@
+    0x00,                                     /* bInterval: ignore for Bulk transfer */
+
+    /* Endpoint IN Descriptor */
+    0x07,                                     /* bLength: Endpoint Descriptor size */
+    USB_DESC_TYPE_ENDPOINT,                   /* bDescriptorType: Endpoint */
+    USBD_PYBRICKS_IN_EP,                      /* bEndpointAddress */
+    USBD_EP_TYPE_BULK,                        /* bmAttributes */
+    LOBYTE(USBD_PYBRICKS_MAX_PACKET_SIZE),    /* wMaxPacketSize: */
+    HIBYTE(USBD_PYBRICKS_MAX_PACKET_SIZE),
+    0x00                                      /* bInterval */
+};
+
+/**
+  * @}
+  */
+
+/** @defgroup USBD_Pybricks_Private_Functions
+  * @{
+  */
+
+/**
+  * @brief  USBD_Pybricks_Init
+  *         Initialize the Pybricks interface
+  * @param  pdev: device instance
+  * @param  cfgidx: Configuration index
+  * @retval status
+  */
+static uint8_t USBD_Pybricks_Init(USBD_HandleTypeDef *pdev, uint8_t cfgidx) {
+    UNUSED(cfgidx);
+    static USBD_Pybricks_HandleTypeDef hPybricks;
+
+    pdev->pClassData = &hPybricks;
+
+    (void)USBD_LL_OpenEP(pdev, USBD_PYBRICKS_IN_EP, USBD_EP_TYPE_BULK, USBD_PYBRICKS_MAX_PACKET_SIZE);
+    pdev->ep_in[USBD_PYBRICKS_IN_EP & 0xFU].is_used = 1U;
+
+    (void)USBD_LL_OpenEP(pdev, USBD_PYBRICKS_OUT_EP, USBD_EP_TYPE_BULK, USBD_PYBRICKS_MAX_PACKET_SIZE);
+    pdev->ep_out[USBD_PYBRICKS_OUT_EP & 0xFU].is_used = 1U;
+
+    /* Init  physical Interface components */
+    ((USBD_Pybricks_ItfTypeDef *)pdev->pUserData[pdev->classId])->Init();
+
+    /* Init Xfer states */
+    hPybricks.TxState = 0U;
+    hPybricks.RxState = 0U;
+
+    (void)USBD_LL_PrepareReceive(pdev, USBD_PYBRICKS_OUT_EP, hPybricks.RxBuffer, USBD_PYBRICKS_MAX_PACKET_SIZE);
+
+    return (uint8_t)USBD_OK;
+}
+
+/**
+  * @brief  USBD_Pybricks_DeInit
+  *         DeInitialize the Pybricks layer
+  * @param  pdev: device instance
+  * @param  cfgidx: Configuration index
+  * @retval status
+  */
+static uint8_t USBD_Pybricks_DeInit(USBD_HandleTypeDef *pdev, uint8_t cfgidx) {
+    UNUSED(cfgidx);
+    uint8_t ret = 0U;
+
+    /* Close EP IN */
+    (void)USBD_LL_CloseEP(pdev, USBD_PYBRICKS_IN_EP);
+    pdev->ep_in[USBD_PYBRICKS_IN_EP & 0xFU].is_used = 0U;
+
+    /* Close EP OUT */
+    (void)USBD_LL_CloseEP(pdev, USBD_PYBRICKS_OUT_EP);
+    pdev->ep_out[USBD_PYBRICKS_OUT_EP & 0xFU].is_used = 0U;
+
+    /* DeInit  physical Interface components */
+    if (pdev->pClassData != NULL) {
+        ((USBD_Pybricks_ItfTypeDef *)pdev->pUserData[pdev->classId])->DeInit();
+        pdev->pClassData = NULL;
+    }
+
+    return ret;
+}
+
+/**
+  * @brief  USBD_Pybricks_Setup
+  *         Handle the Pybricks specific requests
+  * @param  pdev: instance
+  * @param  req: usb requests
+  * @retval status
+  */
+static uint8_t USBD_Pybricks_Setup(USBD_HandleTypeDef *pdev,
+    USBD_SetupReqTypedef *req) {
+    USBD_Pybricks_HandleTypeDef *hPybricks = (USBD_Pybricks_HandleTypeDef *)pdev->pClassData;
```

Comment:
```suggestion
    USBD_Pybricks_HandleTypeDef *hPybricks = pdev->pClassData;
```

cast is not needed since `pClassData` is `void *`